site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/ParticipantFilterHandler.java

```diff
@@ -0,0 +1,263 @@
+    public static Collection<Participant> prepareParticipantsWithFilter( String siteID, List<String> providerCourseIDs, String selectedFilter )
+    {
+        if( StringUtils.isNotBlank( selectedFilter ) )
+        {
+            // Identify the selected filter type and ID
+            String[] parts = selectedFilter.split( "]" );
+            String filterType = identifyFilterType( selectedFilter );
+            String filterID = SiteConstants.PARTICIPANT_FILTER_TYPE_ALL.equals( filterType ) ? "" : parts[1];
+
+            // Get the participants using the selected filter
+            return SiteParticipantHelper.prepareParticipants( siteID, providerCourseIDs, filterType, filterID );
+        }
+        else
+        {
+            return SiteParticipantHelper.prepareParticipants( siteID, providerCourseIDs );
+        }
+    }
+
+    /**
+     * Puts the user's selected filter choice into the context for the velocity template.
+     *
+     * @param state
+     * @param context
+     */
+    public static void putSelectedFilterIntoContext( SessionState state, Context context )
+    {
+        String selectedFilter = (String) state.getAttribute( SiteAction.STATE_SITE_PARTICIPANT_FILTER );
+        if( StringUtils.isBlank( selectedFilter ) )
+        {
+            selectedFilter = SiteConstants.PARTICIPANT_FILTER_TYPE_ALL;
+        }
+        context.put( "selectedFilter", selectedFilter );
+    }
+
+    /**
+     * Puts the user's selected filter choice into the state for retrieval.
+     * 
+     * @param data
+     */
+    public static void putSelectedFilterIntoState( RunData data )
+    {
+        SessionState state = ((JetspeedRunData) data).getPortletSessionState( ((JetspeedRunData) data).getJs_peid() );
+        String filter = StringUtils.trimToNull( Validator.escapeHtml( data.getParameters().getString( "view" ) ) );
+        if( StringUtils.isNotBlank( filter ) )
+        {
+            state.setAttribute( SiteAction.STATE_SITE_PARTICIPANT_FILTER, filter );
+        }
+        else
+        {
+            state.removeAttribute( SiteAction.STATE_SITE_PARTICIPANT_FILTER );
+        }
+    }
+
+    /**
+     * Identify the filter type (all, section, role, or group) based on the selected filter option.
+     *
+     * @param selectedFilter the option selected by the user
+     * @return the identified filter type
+     */
+    private static String identifyFilterType( String selectedFilter )
+    {
+        if( selectedFilter.contains( SiteConstants.PARTICIPANT_FILTER_TYPE_GROUP ) )
+        {
+            return SiteConstants.PARTICIPANT_FILTER_TYPE_GROUP;
+        }
+        else if( selectedFilter.contains( SiteConstants.PARTICIPANT_FILTER_TYPE_SECTION ) )
+        {
+            return SiteConstants.PARTICIPANT_FILTER_TYPE_SECTION;
+        }
+        else if( selectedFilter.contains( SiteConstants.PARTICIPANT_FILTER_TYPE_ROLE ) )
+        {
+            return SiteConstants.PARTICIPANT_FILTER_TYPE_ROLE;
+        }
+        else
```

Comment:
Doesn't really need to be in the else {}, could just return at the end of the method. Makes it slightly shorter.